packages/fhir-router/src/fhirrouter.ts

```diff
@@ -43,7 +43,13 @@ export type FhirRequestConfig = {
   transactions?: boolean;
 };
 
-export type FhirResponse = [OperationOutcome] | [OperationOutcome, Resource];
+export type ContentTypeOverride = string;
+
+// For v4, consider changing FhirResponse from an Array of variable length to an object with optional fields
+export type FhirResponse =
+  | [OperationOutcome]
+  | [OperationOutcome, Resource]
+  | [OperationOutcome, Resource, ContentTypeOverride];
```

Comment:
For future compatibility, how would you feel about making the 3rd arg into an object such as `ExtendedOptions`?  It seems likely that we will want to add more stuff here over time (http headers, cache control, etc)

Reply:
I think we were considering refactoring `FhirResponse` completely for v4 (making it an object)

Reply:
Ah, that makes even more sense.